tests/cephfs/cephfs_nfs/test_cephfs_nfs_ha_snapshot_lifecycle.py

```diff
@@ -0,0 +1,342 @@
+import datetime
+import json
+import secrets
+import string
+import time
+import traceback
+
+from ceph.ceph import CommandFailed
+from ceph.parallel import parallel
+from tests.cephfs.cephfs_utilsV1 import FsUtils as FsUtilsV1
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    CEPH-83574009 - On a A/A NFS-HA cluster, ensure all existing snapshots are listed within the nfs export
+    created on top of subvolume.
+    Steps:
+    1. Create the NFS cluster with the --ingress flag:
+        Syntax
+        ceph nfs cluster create CLUSTER-ID [PLACEMENT] [--port PORT_NUMBER] [--ingress --virtual-ip IP_ADDRESS]
+    2. Validate the Service is UP
+    3. Create subgroup and 2 subvolumes within
+    4. Create 2 Exports one for each subvolume and perform NFS mount on subvolumes.
+    5. Run IOs on both subvolumes for test duration
+    6. On both subvolumes, Create 5 snapshots at minutely interval.
+    7. Verify all 5 snapshots are listed in mount points on both volumes. Run Read IO on latest snapshot dir.
+    8. Get active NFS node. While IO in-progress, run HA failover by rebooting active nfs node.
+    9. Verify failover to new node succeeds. Verify snapshots on both subvolumes are accessible after failover.
+    10.Run write IO on subvolumes. Create 2 new snapshots and verify.Perform Read IO on latest snapshot dir.
+    11. Cleanup
+    """
+    try:
+        tc = "CEPH-83574009"
+        log.info(f"Running cephfs {tc} test case")
+        config = kw["config"]
+        build = config.get("build", config.get("rhbuild"))
+        fs_util = FsUtilsV1(ceph_cluster)
+        clients = ceph_cluster.get_ceph_objects("client")
+        client1 = clients[0]
+        fs_util.prepare_clients(clients, build)
+        fs_util.auth_list(clients)
+        nfs_servers = ceph_cluster.get_ceph_objects("nfs")
+        fs_name = "cephfs"
+        nfs_name = "cephfs-nfs"
+        virtual_ip = "10.8.128.100"
+        subnet = "21"
+
+        log.info("Checking Pre-requisites")
+        if not clients:
+            log.info(
+                f"This test requires minimum 1 client nodes.This has only {len(clients)} clients"
+            )
+            return 1
+
+        client1.exec_command(sudo=True, cmd="ceph mgr module enable nfs")
+        log.info("Creating NFS HA cluster with --ingress flag")
+
+        client1.exec_command(
+            sudo=True,
+            cmd=f'ceph nfs cluster create {nfs_name} "3 {nfs_servers[0].node.hostname} '
```

Comment:
We should always have the count lesser than the available nodes.
It should be "2 {node1} {node2} {node3}"